File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys

nums = int(sys.stdin.readline())
stack = []
ans = []
cnt = 1
chk = True

for i in range(nums):
    num = int(sys.stdin.readline())

    while cnt <= num:       #입력 받은 숫자까지 스택에 push
        stack.append(cnt)
        ans.append('+')
        cnt += 1
    if stack[-1] == num:    #스택의 맨 위와 같다면 pop
        stack.pop()
        ans.append('-')
    else:                   #불가능하므로 NO
        chk = False
        break

if not chk:
    print('NO')
else:
    for a in ans:
        print(a)

File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		Stack<Integer> stack = new Stack<>();

		int n = Integer.parseInt(br.readLine());
		int cnt = 1;
		boolean chk = true;

		for(int i = 0; i < n; i++) {
			int num = Integer.parseInt(br.readLine());

			while(cnt <= num){
				stack.push(cnt);
				sb.append("+\n");
				cnt++;
			}

			if(stack.peek() == num){
				stack.pop();
				sb.append("-\n");
			} else {
				chk = false;
				break;
			}

		}
		if(!chk)
			System.out.println("NO");
		else
			System.out.println(sb);
	}

}